lib/core/elastic_connector_actions.rb

```diff
@@ -247,13 +237,9 @@ def ensure_job_index_exists
         ensure_index_exists("#{JOB_INDEX}-v1", system_index_body(:alias_name => JOB_INDEX, :mappings => mappings))
       end
 
-      def update_connector_field(connector_id, field_name, value)
-        body = {
-          :doc => {
-            field_name => value
-          }
-        }
-        client.update(:index => CONNECTORS_INDEX, :id => connector_id, :body => body)
+      def update_connector_fields(connector_id, doc = {})
```

Comment:
I'm not 100% sure about the name here - should it just be `update_connector`?

This method can also be reused in other methods throughout the file, for example in `update_connector_status` and such, while it can also be outside of scope of this PR

Reply:
> This method can also be reused in other methods throughout the file, for example in `update_connector_status` and such, while it can also be outside of scope of this PR

Good point, I will make the change. It is also reused now by methods like `update_connector_configuration`. The previous method is called `update_connector_field`, which only allow updating one field in one request.

> I'm not 100% sure about the name here - should it just be `update_connector`?

Because this method is heavily used by other methods in `ElasticConnectorActions`, I think it makes more sense to keep it `update_connector_fields`